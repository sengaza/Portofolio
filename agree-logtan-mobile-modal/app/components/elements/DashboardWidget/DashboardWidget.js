import React, { Component } from 'react';
import LinearGradient from 'react-native-linear-gradient';
import { View, Text, TouchableHighlight } from 'react-native';
import I18n from '../../../i18n';
import styles from './DashboardWidget.styles';
import { toTitleCase, noop } from '../../../utils';
import { COLOR_PAGE_BACKGROUND } from '../../../styles';
import SvgArrowRightBlue from '../../../../assets/svgs/ArrowRightBlue';

class DashboardWidget extends Component {
  _renderContent = ({ title = '', content = '' }) => {
    const words = I18n.t(title).split(' ');
    const [head, ...tail] = words;
    const name = toTitleCase(`${head}\r\n${tail.join(' ')}`);
    const number = `${content}`;
    return (
      <View style={styles.content} key={name}>
        <Text style={styles.top}>{number}</Text>
        <Text style={styles.bottom}>{name}</Text>
      </View>
    );
  };

  _renderCardHeader = ({ title = '' }) => {
    const { buttonAction } = this.props;
    return (
      <View style={styles.cardHeader} onPress={buttonAction}>
        <View style={styles.headerWrapper}>
          <Text style={styles.header}>{`${title}`}</Text>
        </View>
        <View style={styles.nextIconWrapper}>
          <SvgArrowRightBlue />
        </View>
      </View>
    );
  };

  _renderCardBody = data => {
    const displayingContent = Object.keys(data).map(title =>
      this._renderContent({ title, content: data[title] })
    );

    return <View style={styles.cardBody}>{displayingContent}</View>;
  };

  render() {
    const {
      cardStyle = {},
      gradient = {},
      displayingData = {},
      cardHeader = {},
      buttonAction = noop
    } = this.props;
    const { start = {}, end = {}, location = [], colors = [] } = gradient;
    return (
      <TouchableHighlight
        activeOpacity={0.2}
        underlayColor={COLOR_PAGE_BACKGROUND}
        style={[styles.container, cardStyle]}
        onPress={buttonAction}
      >
        <LinearGradient
          style={styles.linearWrapper}
          start={start}
          end={end}
          locations={location}
          colors={colors}
        >
          {this._renderCardHeader(cardHeader)}
          {this._renderCardBody(displayingData)}
        </LinearGradient>
      </TouchableHighlight>
    );
  }
}

export default DashboardWidget;
