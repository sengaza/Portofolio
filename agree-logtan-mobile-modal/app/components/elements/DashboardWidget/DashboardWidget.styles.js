import { StyleSheet } from 'react-native';
import {
  COLOR_GREY_90,
  FONT_HEADLINE4,
  FONT_OVERLINE1,
  FONT_LABEL1,
  COLOR_BASE_PRIMARY_DARK
} from '../../../styles/';

const fontWidgetHeader = {
  ...FONT_LABEL1,
  color: COLOR_BASE_PRIMARY_DARK,
  textAlign: 'left'
};

const fontWidgetContentTitle = {
  opacity: 1,
  ...FONT_HEADLINE4,
  letterSpacing: 0,
  textAlign: 'center',
  color: COLOR_GREY_90
};

const fontWidgetContentBody = {
  opacity: 1,
  ...FONT_OVERLINE1,
  textAlign: 'center',
  color: COLOR_GREY_90
};

export default StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'center',
    elevation: 5,
    marginVertical: 10,
    marginHorizontal: 20,
    minHeight: 130,
    minWidth: 319,
    borderWidth: 0
  },
  linearWrapper: {
    flex: 1,
    alignSelf: 'center',
    flexDirection: 'column',
    borderWidth: 0,
    borderRadius: 7
  },
  cardHeader: {
    flex: 1,
    flexDirection: 'row',
    borderRadius: 2,
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  nextIconWrapper: {
    justifyContent: 'flex-end',
    alignItems: 'center',
    margin: 20
  },
  nextIcon: {
    // alignSelf: 'flex-end',
    // resizeMode: 'stretch',
    // height: '100%',
    // width: '30%'
  },
  headerWrapper: {
    width: '70%',
    marginLeft: 20,
    marginRight: 20
  },
  header: {
    ...fontWidgetHeader
  },
  cardBody: {
    flex: 6,
    marginLeft: 20,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  content: {
    flex: 2,
    marginTop: -20,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'flex-start'
  },
  top: {
    ...fontWidgetContentTitle,
    flex: 1,
    textAlign: 'center',
    textAlignVertical: 'bottom'
  },
  bottom: {
    ...fontWidgetContentBody,
    flex: 1,
    textAlign: 'left',
    textAlignVertical: 'top'
  }
});
