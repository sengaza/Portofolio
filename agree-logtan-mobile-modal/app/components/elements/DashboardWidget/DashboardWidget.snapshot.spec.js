import { assertSnapshots } from '../../../utils/TestUtils/snapshot';
import DashboardWidget from './DashboardWidget';

jest.mock('./DashboardWidget.styles', () => ({}));
jest.mock('Text', () => 'Text');

describe('DashboardWidget snapshot testing', () => {
  const configs = [
    {
      state: {},
      props: {
        gradient: {},
        displayingData: {
          assignedToLand: undefined
        }
      },
      desc: 'renders default props'
    },
    {
      state: {},
      props: {
        gradient: {},
        displayingData: {
          assignedToLand: 1
        }
      },
      desc: 'renders default props with displayingData'
    },
    {
      state: {},
      props: {},
      desc: 'renders without props'
    }
  ];
  assertSnapshots(DashboardWidget, configs);
});
