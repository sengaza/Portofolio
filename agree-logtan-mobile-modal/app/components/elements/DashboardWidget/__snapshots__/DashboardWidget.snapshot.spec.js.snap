// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardWidget snapshot testing renders default props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <BVLinearGradient
    colors={Array []}
    endPoint={Object {}}
    locations={Array []}
    startPoint={Object {}}
    style={undefined}
  >
    <View
      onPress={undefined}
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Image
          className="header-icon"
          resizeMode="contain"
          source={null}
          style={undefined}
        />
      </View>
      <View
        className="header-title"
        style={undefined}
      >
        <Text
          allowFontScaling={true}
          style={undefined}
        >
          
        </Text>
      </View>
      <View
        className="header-button"
        style={undefined}
      >
        <Text
          allowFontScaling={true}
          style={undefined}
        >
          total: 0
        </Text>
      </View>
    </View>
    <View
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Text
          style={undefined}
        >
          
        </Text>
        <Text
          style={undefined}
        >
          Assignedtoland

        </Text>
      </View>
    </View>
  </BVLinearGradient>
</View>
`;

exports[`DashboardWidget snapshot testing renders default props with displayingData 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <BVLinearGradient
    colors={Array []}
    endPoint={Object {}}
    locations={Array []}
    startPoint={Object {}}
    style={undefined}
  >
    <View
      onPress={undefined}
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Image
          className="header-icon"
          resizeMode="contain"
          source={null}
          style={undefined}
        />
      </View>
      <View
        className="header-title"
        style={undefined}
      >
        <Text
          allowFontScaling={true}
          style={undefined}
        >
          
        </Text>
      </View>
      <View
        className="header-button"
        style={undefined}
      >
        <Text
          allowFontScaling={true}
          style={undefined}
        >
          total: 0
        </Text>
      </View>
    </View>
    <View
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Text
          style={undefined}
        >
          1
        </Text>
        <Text
          style={undefined}
        >
          Assignedtoland

        </Text>
      </View>
    </View>
  </BVLinearGradient>
</View>
`;

exports[`DashboardWidget snapshot testing renders without props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <BVLinearGradient
    colors={Array []}
    endPoint={Object {}}
    locations={Array []}
    startPoint={Object {}}
    style={undefined}
  >
    <View
      onPress={undefined}
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Image
          className="header-icon"
          resizeMode="contain"
          source={null}
          style={undefined}
        />
      </View>
      <View
        className="header-title"
        style={undefined}
      >
        <Text
          allowFontScaling={true}
          style={undefined}
        >
          
        </Text>
      </View>
      <View
        className="header-button"
        style={undefined}
      >
        <Text
          allowFontScaling={true}
          style={undefined}
        >
          total: 0
        </Text>
      </View>
    </View>
    <View
      style={undefined}
    />
  </BVLinearGradient>
</View>
`;
